src/System.Private.ServiceModel/src/Internals/System/Runtime/TaskHelpers.cs

```diff
@@ -179,16 +179,121 @@ public static void ToApmEnd(this IAsyncResult iar)
             }
         }
 
-        // Task.GetAwaiter().GetResult() calls an internal variant of Wait() which doesn't wrap exceptions in
-        // an AggregateException.
-        public static void WaitForCompletion(this Task task)
+        public static void RunSyncWithParams<T1>(this Func<T1, Task> function, T1 param1)
```

Comment:
Nit: not required, but it would be great to have a comment on the first one of these that explains what is going on just like your PR description.